src/mesh/MeshPacketQueue.cpp

```diff
@@ -16,13 +16,9 @@ bool CompareMeshPacketFunc(const meshtastic_MeshPacket *p1, const meshtastic_Mes
 {
     assert(p1 && p2);
     auto p1p = getPriority(p1), p2p = getPriority(p2);
-
     // If priorities differ, use that
-    // for equal priorities, order by id (older packets have higher priority - this will briefly be wrong when IDs roll over but
-    // no big deal)
-    return (p1p != p2p)
-               ? (p1p < p2p)                                                 // prefer bigger priorities
-               : ((p1->id & ID_COUNTER_MASK) >= (p2->id & ID_COUNTER_MASK)); // Mask to counter portion, prefer smaller packet ids
+    // for equal priorities, order stays as enqueued
+    return (p1p > p2p); // prefer bigger priorities
```

Comment:
Indeed it has been wrong the whole time!

I still think that if the priorities are equal, we should prioritize other node’s packets above those of ourselves for the reasons I laid out here: https://github.com/meshtastic/firmware/pull/4561#issuecomment-2313059473. Text messages have high priority now, so it should not really influence normal messaging on a node that is busy forwarding other people’s packets. Also if another node rebroadcasts before you, other people's messages will be removed anyway. 

I think this can be done by: 
`return (p1p != p2p) ? (p1p > p2p) : (getFrom(p1) != nodeDB->getNodeNum() && getFrom(p2) == nodeDB->getNodeNum())`.

Reply:
I'll make the change and test it.